pkg/storage/tsdb/config.go

```diff
@@ -206,7 +206,7 @@ func (cfg *TSDBConfig) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.HeadChunksWriteQueueSize, "blocks-storage.tsdb.head-chunks-write-queue-size", 0, "The size of the write queue used by the head chunks mapper. Lower values reduce memory utilisation at the cost of potentially higher ingest latency. Value of 0 switches chunks mapper to implementation without a queue.")
 	f.BoolVar(&cfg.NewChunkDiskMapper, "blocks-storage.tsdb.new-chunk-disk-mapper", false, "Temporary flag to select whether to use the new (used in upstream Prometheus) or the old (legacy) chunk disk mapper.")
 	f.BoolVar(&cfg.IsolationEnabled, "blocks-storage.tsdb.isolation-enabled", false, "[Deprecated] Enables TSDB isolation feature. Disabling may improve performance.")
-	f.DurationVar(&cfg.OOOAllowance, "blocks-storage.tsdb.ooo-allowance", 0*time.Second, "Allow samples to be this old for out-of-order.  Supported units: h, m, s.")
+	f.DurationVar(&cfg.OOOAllowance, "blocks-storage.tsdb.ooo-allowance", 0*time.Second, "Allow samples to be this old for out-of-order.  Supported units: h, m, s. If the value is non-zero, then storage.tsdb.allow-overlapping-blocks will be set to true since out-of-order data can potentially overlap with other data.")
```

Comment:
The old comment is specific to Prometheus flag. No flag exists for Mimir. So we need a reword.
```suggestion
	f.DurationVar(&cfg.OOOAllowance, "blocks-storage.tsdb.ooo-allowance", 0*time.Second, "Allow samples to be this old for out-of-order.  Supported units: h, m, s. If the value is non-zero, then all TSDBs will be enabled to handle overlapping data.")
```